src/control/lib/control/pool.go

```diff
@@ -492,7 +496,15 @@ func (pqr *PoolQueryResp) MarshalJSON() ([]byte, error) {
 		aux.DisabledRanks = &ranks
 	}
 
-	return json.Marshal(&aux)
+	auxJSON, err := json.Marshal(&aux)
+	if err != nil {
+		return nil, err
+	}
+
+	// Kinda gross, but needed to merge the embedded struct's MarshalJSON
+	// output with this one's.
+	piJSON[0] = ','
+	return append(auxJSON[:len(auxJSON)-1], piJSON...), nil
```

Comment:
I'm a little confused. If PoolInfo marshals cleanly, why is it necessary to separate it out and tack it on, rather than including it as part of the aux struct?

Reply:
It's due to one of the quirks of using embedded types. The `MarshalJSON` method on `daos.PoolInfo` is [promoted](https://go.dev/ref/spec#Struct_types), so the logic in `PoolQueryResp`'s `MarshalJSON` is never used.